lib/req/steps.ex

```diff
@@ -171,23 +174,17 @@ defmodule Req.Steps do
     put_new_header(request, "authorization", "Basic #{value}")
   end
 
-  @doc """
-  Sets request authentication for a matching host from a netrc file.
-
-  ## Examples
-
-      iex> Req.get!("https://httpbin.org/basic-auth/foo/bar").status
-      401
-      iex> Req.get!("https://httpbin.org/basic-auth/foo/bar", netrc: true).status
-      200
-      iex> Req.get!("https://httpbin.org/basic-auth/foo/bar", netrc: "/path/to/custom_netrc").status
-      200
+  def auth(request, :netrc) do
+    directory = System.get_env("NETRC", System.user_home!())
+    path = Path.join(directory, ".netrc")
```

Comment:
https://www.gnu.org/software/inetutils/manual/html_node/The-_002enetrc-file.html is a vague whether `NETRC` holds the path directory (with `.netrc`) or the path to the file.

I think treating it as the path gives us more flexibility. Seems that's what Requests does too. https://github.com/psf/requests/pull/5643/files#diff-f131fe49192d62f76956ec2fdce8928baaa3cd8cbc070e4d5ac1981642bdadc9R217

```suggestion
    directory = System.get_env("NETRC") || Path.join(System.user_home!(), ".netrc")
```